ddtrace/internal/ci_visibility/recorder.py

```diff
@@ -306,12 +321,19 @@ def _fetch_tests_to_skip(self):
         for item in parsed["data"]:
             if item["type"] == TEST_SKIPPING_LEVEL and "suite" in item["attributes"]:
                 module = item["attributes"].get("configurations", {}).get("test.bundle", "").replace(".", "/")
-                self._test_suites_to_skip.append(
-                    "/".join((module, item["attributes"]["suite"])) if module else item["attributes"]["suite"]
-                )
+                path = "/".join((module, item["attributes"]["suite"])) if module else item["attributes"]["suite"]
+
+                if TEST_SKIPPING_LEVEL == SUITE:
+                    self._test_suites_to_skip.append(path)
+                else:
+                    self._tests_to_skip[path].append(item["attributes"]["name"])
 
-    def _should_skip_path(self, path):
-        return os.path.relpath(path) in self._test_suites_to_skip
+    def _should_skip_path(self, path, name):
+        if _get_test_skipping_level() == SUITE:
```

Comment:
Maybe this test-skipping level should be an argument to this function. It arguably shouldn't change at all from the time the plugin starts, let alone between calls to `_should_skip_path()`.